apps/bbsync/tests/test_query.py

```diff
@@ -201,18 +201,77 @@ def test_generate_summary_removed_cve(self):
             (WorkflowModel.WorkflowState.PRE_SECONDARY_ASSESSMENT, "vulnerability"),
             (WorkflowModel.WorkflowState.SECONDARY_ASSESSMENT, "vulnerability"),
             (WorkflowModel.WorkflowState.DONE, "vulnerability"),
-            (WorkflowModel.WorkflowState.REJECTED, "vulnerability"),
+            (WorkflowModel.WorkflowState.REJECTED, "vulnerability-draft"),
         ],
     )
-    def test_generate_component(self, workflow_state, result):
+    def test_generate_component_before_bz_sync(self, workflow_state, result):
         """
-        test generating of component
+        Test that component is set to "vulnerability-draft" in NEW and REJECTED
+        workflow states and to "vulnerability" in other states when the flaw
+        hasn't been synced to bugzilla yet.
         """
         flaw = FlawFactory(workflow_state=workflow_state)
 
         bbq = FlawBugzillaQueryBuilder(flaw)
         assert bbq.query["component"] == result
 
+    @pytest.mark.parametrize(
+        "workflow_state,result",
+        [
+            (WorkflowModel.WorkflowState.NOVALUE, "vulnerability"),
+            (WorkflowModel.WorkflowState.NEW, "vulnerability-draft"),
+            (WorkflowModel.WorkflowState.TRIAGE, "vulnerability"),
+            (WorkflowModel.WorkflowState.PRE_SECONDARY_ASSESSMENT, "vulnerability"),
+            (WorkflowModel.WorkflowState.SECONDARY_ASSESSMENT, "vulnerability"),
+            (WorkflowModel.WorkflowState.DONE, "vulnerability"),
+            (WorkflowModel.WorkflowState.REJECTED, "vulnerability-draft"),
+        ],
+    )
+    def test_generate_component_after_bz_sync_with_draft_component(
+        self, workflow_state, result
+    ):
+        """
+        Test that component is set to "vulnerability-draft" in NEW and REJECTED
+        workflow states and to "vulnerability" in other states when the flaw
+        has been synced to bugzilla already with component previously set to
+        "vulnerability-draft".
+        """
+        flaw = FlawFactory(
+            workflow_state=workflow_state,
+            meta_attr={"bz_id": "123", "bz_component": "vulnerability-draft"},
+        )
+
+        bbq = FlawBugzillaQueryBuilder(flaw)
+        assert bbq.query["component"] == result
+
+    @pytest.mark.parametrize(
+        "workflow_state,result",
+        [
+            (WorkflowModel.WorkflowState.NOVALUE, "vulnerability"),
+            (WorkflowModel.WorkflowState.NEW, "vulnerability"),
+            (WorkflowModel.WorkflowState.TRIAGE, "vulnerability"),
+            (WorkflowModel.WorkflowState.PRE_SECONDARY_ASSESSMENT, "vulnerability"),
+            (WorkflowModel.WorkflowState.SECONDARY_ASSESSMENT, "vulnerability"),
+            (WorkflowModel.WorkflowState.DONE, "vulnerability"),
+            (WorkflowModel.WorkflowState.REJECTED, "vulnerability"),
```

Comment:
No need to parametrize a constant (i.e. `vulnerability`). You can use `assert bbq.query["component"] == "vulnerability"` directly in the test.

Reply:
Thanks, fixed using your suggestion.